src/Dapr.Workflow/WorkflowServiceCollectionExtensions.cs

```diff
@@ -57,7 +59,19 @@ public static class WorkflowServiceCollectionExtensions
 
                 if (TryGetGrpcAddress(out string address))
                 {
-                    builder.UseGrpc(address);
+                    var apiToken = Environment.GetEnvironmentVariable("DAPR_API_TOKEN");
+                    if (!string.IsNullOrEmpty(apiToken))
+                    {
+                        // serviceCollection.ConfigureDurableGrpcClient(apiToken);
+                        HttpClient client = new HttpClient();
+                        client.DefaultRequestHeaders.Add("Dapr-Api-Token", apiToken);
+                        builder.UseGrpc(CreateChannel(client));
```

Comment:
The `address` variable should be passed to the builder, otherwise if users configured Dapr to use a non-localhost endpoint, this will not be propagated correctly to the underlying Durable Task client. Good find by @alicejgibbons